### PR TITLE
gh-120319: Include typecheck for `maxsize` on `Queue` initialization.

### PR DESCRIPTION
**Pull request by pygeek**
* Issue: gh-120319

### CODE REVIEW

**Comment by pygeek** [comment]
I believe some changes need to be made to the C API. Would you say that `_interpqueuesmodule.c` is the only file that needs to be updated? I don't see any other modules referencing `maxsize`. @rhettinger

Also, once I make my changes do I need to run the clinic tool against the file?

**Comment by pygeek** [comment]
I added further context to this PR here: https://github.com/python/cpython/issues/120319#issuecomment-2163996122

File: Lib/test/test_queue.py [hunk 1]
```diff
@@ -663,11 +663,34 @@ def setUp(self):
...
+                self.queue.Queue(maxsize=val)
+
+    def test_maxsize_int_type(self):
+        try:
+            self.queue.Queue(maxsize=1)
+        except TypeError:
+            self.fail("Queue(maxsize=1) unexpectedly raised TypeError.")
+
+
+
```
Comment (nineteendo) [inline Lib/test/test_queue.py]:
```suggestion

```

File: Lib/test/test_queue.py [hunk 2]
```diff
@@ -663,11 +663,34 @@ def setUp(self):
...
 
+    def test_maxsize_non_int_type(self):
+        for val in (None, "test", 1.0):
+            with self.assertRaises(TypeError):
+                self.queue.Queue(maxsize=val)
+
+    def test_maxsize_int_type(self):
+        try:
+            self.queue.Queue(maxsize=1)
+        except TypeError:
```
Comment (sobolevn) [inline Lib/test/test_queue.py]: this is not needed, `TypeError` would fail the test by itself.

**Comment by rhettinger** [comment]
Thanks for the suggestion, but I don't think this is needed. The *maxsize* argument has been around for a very long time and this has never been an issue.

**Pull request closed without merging**